vars/commonPipelineEnvironment.groovy

```diff
@@ -216,7 +216,7 @@ class commonPipelineEnvironment implements Serializable {
                 if(value in CharSequence) {
                     script.writeFile file: fileName, text: value
                 } else {
-                    script.writeFile file: fileName, text: groovy.json.JsonOutput.toJson(value)
+                    script.writeFile file: fileName + ".json", text: groovy.json.JsonOutput.toJson(value)
```

Comment:
Above we have a `fileExists` for `fileName`, but here we write another file (mind the `.json` appendix). IMO it would be better to add the `.json` suffix above when the variable is declared.

Reply:
Why do we check the existence here? WriteToDisk should always write the newest data.

> Is this change required in order to fix the issue denoted in the title ("correct writing of complex types ...")

Yes, and only this.

Reply:
My main point was: above we check for a file and below we write another file (above we check without appendix and below we write with appendix). Is that fine?

Next question would be: and when the file already exists? In that case we don't update. We only write the file it is not already present.  Is that the intended way? What if the content has been changed in the meantime. In that case we keep a file with some older content ?

Reply:
extracted the change to https://github.com/SAP/jenkins-library/pull/2500